src/core_plugins/kibana/public/dashboard/grid/dashboard_grid.js

```diff
@@ -134,7 +134,7 @@ export class DashboardGrid extends React.Component {
 
   onLayoutChange = (layout) => {
     const { onPanelsUpdated } = this.props;
```

Comment:
It's debatable which would be clearer, but this forEach is being used to do a reduce. I find that reduce explicitly states the purpose of the loop, so that's preferable. On the other hand, reduce is often harder to follow than a simple loop. Here's a more functional alternative, but I say pick whichever approach you think is clearest:

```js
onLayoutChange = (layout) => {
  const { onPanelsUpdated } = this.props;
  const updatedPanels = layout.reduce((acc, panelLayout) => {
    return Object.assign(acc, {
      [panelLayout.i]: {
        panelIndex: panelLayout.i,
        gridData: _.pick(panelLayout, ['x', 'y', 'w', 'h', 'i']),
      },
    });
  }, {});
  onPanelsUpdated(updatedPanels);
};
```